src/App.tsx

```diff
@@ -1,109 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+const getItemNum = (item: string): number => {
+  const idx = item.indexOf(' ');
+  return +item.slice(idx);
+};
+
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+
+  const total: number = items.length;
+  const lastItemIndex: number = currentPage * perPage;
+  const firstItemIndex: number = lastItemIndex - perPage;
+  const itemsOnPage: string[] = items.slice(firstItemIndex, lastItemIndex)
+
+
+
+  const onOptionChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  }
+
+
+  const onPageChange = (pageNumber: number) => {
+    setCurrentPage(pageNumber);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${getItemNum(itemsOnPage[0])} - ${getItemNum(itemsOnPage[itemsOnPage.length - 1])} of ${total})`}
```

Comment:
Potential issue: `itemsOnPage` could be empty, which would cause `getItemNum(itemsOnPage[0])` and `getItemNum(itemsOnPage[itemsOnPage.length - 1])` to throw an error. Consider adding a check to ensure `itemsOnPage` is not empty before accessing its elements.